lib/context.js

```diff
@@ -56,17 +57,49 @@ function Context ({
   this[kFourOhFourContext] = null
   this.attachValidation = attachValidation
   this[kReplySerializerDefault] = replySerializer
-  this.schemaErrorFormatter = schemaErrorFormatter || server[kSchemaErrorFormatter] || defaultSchemaErrorFormatter
+  this.schemaErrorFormatter =
+    schemaErrorFormatter ||
+    server[kSchemaErrorFormatter] ||
+    defaultSchemaErrorFormatter
   this[kRouteByFastify] = isFastify
 
   this[kRequestValidateWeakMap] = null
   this[kReplySerializeWeakMap] = null
   this.validatorCompiler = validatorCompiler || null
   this.serializerCompiler = serializerCompiler || null
 
+  // Route + Userland configurations for the route
+  this[kPublicRouteContext] = getPublicRouteContext(this)
+
   this.server = server
 }
 
+function getPublicRouteContext (context) {
+  return Object.create(null, {
+    schema: {
+      enumerable: true,
+      get () {
+        return context.schema
+      }
+    },
+    config: {
+      enumerable: true,
+      get () {
+        return context.config
+      }
+    },
+    logLevel: {
```

Comment:
I was playing a bit and might be beneficial to have some sort of single place where to look for specific settings of the route definition. I was thinking of a single `read-only` approach but wanted to get your thoughts on this, as might/might-not have sense. wdyt?

Reply:
I would not say it is non-sense, it is just something nobody asked for so I would work on it in a separate PR if necessary instead of adding too much code on this one

Reply:
Has sense, let's keep it clean 👍 
Done in 2c188371